### PR TITLE
Clean up infomax

### PR DESCRIPTION
**Pull request by cbrnr**
Specifically, the `extended` argument now really defaults to `True` (the docstring claimed this before even though it was really `False`). I prettified the docstring and comments, added the defaults in the docstring, and replaced `if extended is True:` with `if extended:`.

### CODE REVIEW

**Comment by mmagnuski** [comment]
LGTM!

**Comment by Eric89GXL** [comment]
LGTM as well, CI error is unrelated

File: mne/preprocessing/infomax_.py [hunk 1]
```diff
@@ -13,94 +13,92 @@
...
+            anneal_deg=60., anneal_step=0.9, extended=True, n_subgauss=1,
+            kurt_size=6000, ext_blocks=1, max_iter=200, random_state=None,
+            blowup=1e4, blowup_fac=0.5, n_small_angle=20, use_bias=True,
+            verbose=None):
+    """Run (extended) Infomax ICA decomposition on raw data.
 
-    based on the publications of Bell & Sejnowski 1995 (Infomax)
-    and Lee, Girolami & Sejnowski, 1999 (extended Infomax)
+    Based on the publications of Bell & Sejnowski, 1995 (Infomax)
+    and Lee, Girolami & Sejnowski, 1999 (extended Infomax).
```
Comment (wmvanvliet) [inline mne/preprocessing/infomax_.py]: while you're at it, could you include the full references to these papers in a `References` section of the docstring?

File: mne/preprocessing/infomax_.py [hunk 2]
```diff
@@ -13,94 +13,92 @@
...
-        The initialized unmixing matrix. Defaults to None. If None, the
-        identity matrix is used.
+    weights : np.ndarray, shape (n_features, n_features)
+        The initialized unmixing matrix.
+        Defaults to None, which means the identity matrix is used.
     l_rate : float
         This quantity indicates the relative size of the change in weights.
-        Note. Smaller learining rates will slow down the procedure.
-        Defaults to 0.010d / alog(n_features ^ 2.0)
+        Note: Smaller learning rates will slow down the ICA procedure.
```
Comment (Eric89GXL) [inline mne/preprocessing/infomax_.py]: this should probably be:
```
... 2.0).

.. note:: Smaller learning ...

Defaults to ...
```

File: mne/preprocessing/infomax_.py [hunk 3]
```diff
@@ -13,94 +13,92 @@
...
+        Defaults to 0.01 / log(n_features ** 2).
     block : int
-        The block size of randomly chosen data segment.
-        Defaults to floor(sqrt(n_times / 3d))
+        The block size of randomly chosen data segments.
+        Defaults to floor(sqrt(n_times / 3)).
```
Comment (Eric89GXL) [inline mne/preprocessing/infomax_.py]: If we're being strictly correct this should be `/ 3.`

**Comment by Eric89GXL** [comment]
A rebase would be good, too, so that CircleCI can render the docs and we can have a look. But really all of these comments we could take care of in another PR if you want.

**Comment by cbrnr** [comment]
Do the docs look OK?

**Comment by mmagnuski** [comment]
* `infomax` is not in the `doc/python_reference.rst` so the docs for infomax are not rendered
* just BTW - why is the file name `infomax_.py`? (why the underscore?)

**Comment by agramfort** [comment]
+1 to add infomax function to python_reference.rst

infomax_.py so if you do `from mne.preprocessing import infomax` you get the function and not the file as a module.

last but not least to me this is an API change without deprecation. Is this really needed?

**Comment by Eric89GXL** [comment]
The docs didn't match behavior, so maybe that as a bug justifies it

**Comment by cbrnr** [comment]
To get the function instead of the module, the module could also be named `infomax.py` with the following line in `__init__.py` within the `preprocessing` package:

    from .infomax import infomax

Maybe a bit less confusing than naming the module `infomax_.py`.

**Comment by cbrnr** [comment]
Do you want me to implement this change in this PR (i.e. rename the module to `infomax.py` and change the import in `__init__.py`)?

**Comment by agramfort** [comment]
Please don't 






On Thu, Aug 11, 2016 at 10:03 AM +0200, "Clemens Brunner" <notifications@github.com> wrote:












Do you want me to implement this change in this PR (i.e. rename the module to infomax.py and change the import in __init__.py)?

**Comment by cbrnr** [comment]
No problem. Anything else I can do in this PR?

**Comment by agramfort** [comment]
please update what's new API section and +1 for merge.

**Pull request merged**